plugins/modules/event.py

```diff
@@ -0,0 +1,221 @@
+    U(https://docs.sensu.io/sensu-go/latest/reference/events/)
+extends_documentation_fragment:
+  - sensu.sensu_go.base
+options:
+  timestamp:
+    description:
+      - Time that the event occurred in seconds since the Unix epoch.
+    type: int
+  state:
+    description:
+      - Target state of the Sensu object.
+    type: str
+    choices: [ 'present', 'absent' ]
+    default: present
+  entity:
+    description:
+      - Entity object returned from M(entity) or M(entity_info).
+    type: dict
+  check:
+    description:
+      - Check object returned from M(check) or M(check_info).
+    type: dict
+  check_response:
+    description:
+      - Information about the status and history of the event.
+        U(https://docs.sensu.io/sensu-go/5.13/reference/events/#check-attributes)
+    type: dict
+    suboptions:
+      duration:
+        description: Command execution time in seconds.
+        type: float
+      executed:
+        description: Time that the check request was executed.
+        type: int
+      history:
+        description: Check status history for the last 21 check executions.
+        type: list
+      issued:
+        description: Time that the check request was issued in seconds since the Unix epoch.
+        type: int
+      last_ok:
+        description: The last time that the check returned an OK status (0) in seconds since the Unix epoch.
+        type: int
+      output:
+        description: The output from the execution of the check command.
+        type: str
+      state:
+        description: The state of the check.
+        choices: [ "passing", "failing", "flapping" ]
+        type: str
+      status:
+        description: Exit status code produced by the check.
+        choices: [ "ok", "warning", "critical", "unknown" ]
+        type: str
+      total_state_change:
+        description: The total state change percentage for the check's history.
+        type: int
+'''
+
+EXAMPLES = '''
+- name: Create a simple entity
+  entity:
+    auth:
+      url: http://localhost:8080
+    name: awesome_entity
+    entity_class: proxy
+  register: entity
+
+- name: Create a simple check
+  check:
+    auth:
+      url: http://localhost:8080
+    name: awesome_check
+    command: echo "Check command"
+    output_metric_handlers:
+      - influxdb
+    subscriptions:
+      - checks
+      - also_checks
+    interval: 30
+  register: check
+
+- name: Create event with minimal parameters
+  event:
+    auth:
+      url: http://localhost:8080
+    entity: "{{ entity.object }}"
+    check: "{{ check.object }}"
+    check_response:
+      output: Check command
+'''
+
+RETURN = '''
+objects:
+  description: list of Sensu events
+  returned: always
+  type: list
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible_collections.sensu.sensu_go.plugins.module_utils import (
+    arguments, errors, utils, debug
+)
+
+
+def do_differ(current, desired):
```

Comment:
Should we add a test or two to the module unit test?

Also, we probably need to decide how to handle structural mismatches. The original comparison function reports those as differences, while custom implementation for this module will raise an exception.

I guess the right way to go would be to detect those mismatches and report them to the user. Most of the time, those mismatches will indicate that Sensu Go made a backward-incompatible change to the API. And we would like to get notified about them with as much information as possible. Any suggestions on how to approach this?

Reply:
Yes of course, I will add some test.

As for the structural changes, in the case of structural mismatch we could throw an error explaining what when wrong and abort the module. Continuing the module is probably not wise since the structure between local and remote is different, and with that kind of API we don't really know what would happen :).
Our integration tests should notify us about the change.

For example if the remote payload is:

```
{"a": "b", "c": "d"}
```
and local is:
```
{"a": "b", "c": {"d": "e"}}
```

In this case we would abort the module and report the error to user with something like this:
`"Remote and local value on key 'c' structurally mismatch"`.

Reply:
Been thinking about this for 10min now, good question :)

The UX we currently offer with our modules does not bring the concept of "diff" to user yet - she doesn't really know that we're even performing GET request for her (prior PUTing). She just knows we're creating entity as she specified - and I think this should be maintained. Following that UX I'm not sure we would want to report structural mismatch back to the user.

How can we debug it then, in case of error?? Well, recently we introduced simple logging mechanism that logs all REST API requests and all responses. In case of backward-incompatible change we would turn this logging on and manually spot the difference. Or, if we don't have access to her environment, she would send the log file to us to examine.

Bottomline, my suggestion is to silently "fail" i.e. to mark the entity as "changed" and proceed to PUT request. In case of incompatible API change the PUT will fail (with 400) and at that point user will reach out to us. Then we bring out the debug-logger procedure. Looking forward to your opinion.

Reply:
I tend to agree with @miha-plesko here: we should make sure that the comparison does not raise an exception on structural mismatches and let the thing error-out later down the line. But I would still add a debug message if we detect the issue when comparing the payloads. This should make it a bit easier to determine the root cause of the issue down the line.